### PR TITLE
Further optimize the code

### PR DESCRIPTION
**Pull request by szepeviktor**
- remove temporary variables
- avoid for loops

### CODE REVIEW

File: src/functions.php [hunk 1]
```diff
@@ -49,17 +44,16 @@ function fetch(string $url, array $options = []): Response
 
     // We extract relevant stream meta data
     $meta = stream_get_meta_data($resource);
-    $rawHeaders = $meta['wrapper_data'];
 
     // We create objects out of that data.
-    $partials = HttpPartialResponse::parseLines($rawHeaders);
+    $partials = HttpPartialResponse::parseLines($meta['wrapper_data']);
+    /** @var HttpPartialResponse $mainPartial */
     $mainPartial = array_pop($partials);
```
Comment (szepeviktor) [inline src/functions.php:51]: Is it sure it'll be a non-empty array?

**Comment by szepeviktor** [comment]
If you like once used variables just close this one with a quick click.

**Comment by mnavarrocarter** [comment]
Thanks @szepeviktor. Fine with me!

**Comment by szepeviktor** [comment]
You're welcome.

**Pull request merged**